polaris-react/src/components/InContextLearning/tests/InContextLearning.test.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import {mountWithApp} from 'tests/utilities';
+
+import {Button} from '../../Button';
+import {InContextLearning} from '../InContextLearning';
+import {InContextLearningContextProvider} from '../components/InContextLearningContext';
+
+describe('<InContextLearning />', () => {
+  const stepOneContent = 'Step one content';
+  const stepTwoContent = 'Step two content';
+  const stepThreeContent = 'Step three content';
+
+  const tutorialOneContent = 'Tutorial one content';
+  const tutorialTwoContent = 'Tutorial two content';
+  const tutorialThreeContent = 'Tutorial three content';
+
+  const StepOneComponent = () => <div>{tutorialOneContent}</div>;
+  const StepTwoComponent = () => <div>{tutorialTwoContent}</div>;
+  const StepThreeComponent = () => <div>{tutorialThreeContent}</div>;
+
+  it('renders and registers all steps and can reverse through them', async () => {
+    const onDismiss = jest.fn();
+
+    const subject = mountWithApp(
+      <InContextLearningContextProvider stepComponents={[
+        <StepOneComponent/>,
+        <StepTwoComponent/>,
+        <StepThreeComponent/>
+      ]}>
+        <InContextLearning.Step stepIndex={0}>
+          <div>{stepOneContent}</div>
+        </InContextLearning.Step>
+        <InContextLearning.Step stepIndex={1}>
+          <div>{stepTwoContent}</div>
+        </InContextLearning.Step>
+        <InContextLearning.Step stepIndex={2}>
+          <div>{stepThreeContent}</div>
+        </InContextLearning.Step>
+        <InContextLearning onDismiss={onDismiss}/>
+      </InContextLearningContextProvider>
+    );
+
+    let closeButton;
```

Comment:
The reason for the button reordering is that the buttons pop in and out based on the number step you are on. We could handle this better with explicit ids or selectors on the components, but I figured that would be better off as its own PR.